src/icalendar/prop/__init__.py

```diff
@@ -1153,8 +1167,14 @@ def __repr__(self):
     def dt(self):
         """Make this cooperate with the other vDDDTypes."""
         return (self.start, (self.duration if self.by_duration else self.end))
+    @property
+    def ics_value(self):
+        if self.by_duration:
+            return (self.start, self.duration)
```

Comment:
This duplicates `.dt`. Please just return `self.dt`.

Also, they all need a type hint in the end. I updated the issue to be clearer on the scope.